applications/FluidDynamicsApplication/custom_processes/two_fluids_inlet_process.cpp

```diff
@@ -0,0 +1,243 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Simon Wenczowski
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "two_fluids_inlet_process.h"
+#include "fluid_dynamics_application_variables.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+TwoFluidsInletProcess::TwoFluidsInletProcess(
+    ModelPart& rModelPart,
+    Parameters& rParameters,
+    Process::Pointer p_dist_proc )
+    : Process(), mrInletModelPart(rModelPart) {
+
+    KRATOS_CHECK_VARIABLE_KEY( DISTANCE )
+    KRATOS_CHECK_VARIABLE_KEY( AUX_DISTANCE )
+    KRATOS_CHECK_VARIABLE_KEY( NODAL_AREA )
+
+    // checking all parameters that define the interface
+    Parameters default_parameters( R"(
+    {
+        "interface_normal"          : [0.0,1.0,0.0],
+        "point_on_interface"        : [0.0,0.25,0.0],
+        "inlet_transition_radius"   : 0.05
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    // finding the complete model of the inlet model part
+    ModelPart& rRootModelPart = mrInletModelPart.GetRootModelPart();
+
+    // setting the parameters to the private data members of the class
+    mInterfaceNormal = rParameters["interface_normal"].GetVector();
+    mInterfacePoint = rParameters["point_on_interface"].GetVector();
+    mInletRadius = rParameters["inlet_transition_radius"].GetDouble();
+
+    // normalization of itnerface normal vector
+    if ( norm_2( mInterfaceNormal ) > 1.0e-7 ){
+        mInterfaceNormal = mInterfacePoint / norm_2( mInterfaceNormal );
+    } else {
+        KRATOS_ERROR << "Error thrown in TwoFluidsInletProcess: 'interface_normal' in 'interface_settings' must not have a norm of 0.0." << std::endl;
+    }
+    // checking to avoid 0 transition radius (necessary for division)
+    if ( mInletRadius < 1.0e-7 ){
+        KRATOS_ERROR << "Error thrown in TwoFluidsInletProcess: 'inlet_transition_radius' in 'interface_settings' must not have a value smaller or equal 0.0." << std::endl;
+    }
+
+    // setting flags for the inlet on nodes and conditions
+    #pragma omp parallel for
+    for (int i_node = 0; i_node < static_cast<int>( mrInletModelPart.NumberOfNodes() ); ++i_node){
+        // iteration over all nodes
+        auto it_nodes = mrInletModelPart.NodesBegin() + i_node;
+        it_nodes->Set( INLET, true );
+    }
+    #pragma omp parallel for
+    for (int i_cond = 0; i_cond < static_cast<int>( mrInletModelPart.NumberOfConditions() ); ++i_cond){
```

Comment:
Same (you can use the method `SetFlag` in the `VariableUtils` to do this operation. It already includes the parallelism pragmas).

Reply:
Yes. Done.